src/DataHandling/Logs/DataDogAudit.php

```diff
@@ -114,28 +170,69 @@ protected function getVersionKey(AuditLog $currentVersion, $diffArray)
 
     /**
      * Method creating diff for given log entry.
-     * @param \DataDog\AuditBundle\Entity\AuditLog $currentVersion
-     * @param array $diffElement current state of diff for this version (one version can consist of more then one log entry)
+     *
+     * @param AuditLog $currentVersion
+     * @param array    $diffElement    current state of diff for this version (one version can consist of more then one log entry)
+     *
      * @return array diff element for given version
      */
     protected function getCurrentVersionElement(AuditLog $currentVersion, array $diffElement)
     {
-        if ($currentVersion->getAction()=='associate') {
-            $diffElement[$this->getColumnNameForAssociation($currentVersion)][self::KEY_ADD][$currentVersion->getTarget()->getFk()] = $currentVersion->getTarget()->getLabel();
-        } else if ($currentVersion->getAction()=='dissociate') {
+        if (empty($this->cache['entityTableName'])) {
+            $this->cache['entityTableName'] = $this->getEntitiesClassMetaData()->getTableName();
+        }
+        $entityTable = $this->cache['entityTableName'];
+        if ($currentVersion->getTbl() !== $entityTable && $currentVersion->getSource()->getTbl() !== $entityTable) {
+            // owning side of ManyToMany association, the side is not this entity
             $columnName = $this->getColumnNameForAssociation($currentVersion);
-            if (!isset($diffElement[$columnName][self::KEY_ADD][$currentVersion->getTarget()->getFk()])) {
-                $diffElement[$columnName][self::KEY_REMOVE][$currentVersion->getTarget()->getFk()] = $currentVersion->getTarget()->getLabel();
-            } else {    // when dissociate and associate on same element that means, that it was before
-                unset($diffElement[$columnName][self::KEY_ADD][$currentVersion->getTarget()->getFk()]);
-                $diffElement[$columnName][self::KEY_UNCHANGED][$currentVersion->getTarget()->getFk()] = $currentVersion->getTarget()->getLabel();
-                if (empty($diffElement[$columnName][self::KEY_ADD])) {
-                    unset($diffElement[$columnName][self::KEY_ADD]);
+            $label = $this->getLabelForAssociation($currentVersion->getSource());
+            if ('insert' === $currentVersion->getAction()) {
```

Comment:
We really need here identical operator? Generally, identical comparison operator ('===') is slower than equal operator ('==') and is used in situation where it is really important (like to differentiate between 0 and false by stripos function)

Reply:
see https://github.com/EmchBerger/cube-common-bundle/pull/28#discussion_r163605067
our style guide tells to use `===` except if there is a reason for NOT doing it.

Is `==` really quicker? `$a == $b` does several steps. It checks if $a is same as $b. If not, it checks if `(string)$a === (string)$b`.

Reply:
Ok, identical operator is really quicker